connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ModifiableResourceAdapter.java

```diff
@@ -71,5 +71,20 @@ public void setRaXmlDeploymentServiceName(ServiceName raXmlDeploymentServiceName
         this.raXmlDeploymentServiceName = raXmlDeploymentServiceName;
     }
 
+    @Override
+    public synchronized int hashCode() {
+        return super.hashCode();
+    }
+
+    @Override
+    public synchronized boolean equals(Object obj) {
+        return super.equals(obj);
+    }
+
+    @Override
+    public synchronized String toString() {
+        return super.toString();
+    }
+
```

Comment:
I think this is fine as a workaround, but I think the upstream thread-safety issue should still be addressed.  WDYT?

Reply:
@pferraro you mean fixing it in ironjacamar? tbh my line of thinking was that ActivationImpl (the parent class) works on collections that are provided to it so theoretically some other implementations could decide to construct it with synchronized collections. But giving a second look, I see that in ActivationImpl creates ArrayLists itself anyway. As a result I agree, I'm going to synchronize the method in IJ code.